pkg/image/api/helper.go

```diff
@@ -145,21 +154,108 @@ func ImageWithMetadata(image Image) (*Image, error) {
 	return &image, nil
 }
 
+func TagValueToTagEvent(repo *ImageRepository, value string) (*TagEvent, error) {
+	if strings.Contains(value, "@") {
+		segs := strings.SplitN(value, "@", 2)
+		if len(segs[1]) == 0 {
+			return nil, fmt.Errorf("%q may not end with a @", value)
+		}
+		ref, err := DockerImageReferenceForRepository(repo)
+		if err != nil {
+			return nil, err
+		}
+		ref.ID = segs[1]
+		return &TagEvent{
+			Created:              util.Now(),
+			DockerImageReference: ref.String(),
+			Image:                ref.ID,
+		}, nil
+	}
+	return LatestTaggedImage(repo, value)
+}
+
+// DockerImageReferenceForRepository returns a DockerImageReference that represents
+// the ImageRepository or false, if no valid reference exists.
+func DockerImageReferenceForRepository(repo *ImageRepository) (DockerImageReference, error) {
+	spec := repo.Status.DockerImageRepository
+	if len(spec) == 0 {
+		spec = repo.DockerImageRepository
+	}
+	if len(spec) == 0 {
+		return DockerImageReference{}, fmt.Errorf("no possible pull spec for %s/%s", repo.Namespace, repo.Name)
+	}
+	return ParseDockerImageReference(spec)
+}
+
 // LatestTaggedImage returns the most recent TagEvent for the specified image
-// repository and tag.
-func LatestTaggedImage(repo ImageRepository, tag string) (*TagEvent, error) {
-	if _, ok := repo.Tags[tag]; !ok {
-		return nil, fmt.Errorf("image repository %s/%s: tag %q not found", repo.Namespace, repo.Name, tag)
+// repository and tag. Will resolve lookups for the empty tag.
```

Comment:
what does "will resolve lookups for the empty tag" mean?  will resolve them to what?  (clearly "latest" as the code stands now..)